distributed/scheduler.py

```diff
@@ -7477,7 +7504,33 @@ def decide_worker(
         for ws in candidates:
             break
     else:
+        group: TaskGroup = ts._group
+        ws = group._last_worker
+
+        total_nthreads = sum(
+            wws._nthreads for wws in candidates
+        )  # TODO get `self._total_threads` from scheduler? Though that doesn't account for worker restrictions.
```

Comment:
Common case, if `len(candidates) < len(self.workers)` ... else `self._total_threads`

Reply:
Yeah. More that this function isn't a method of `SchedulerState`, so we'll have to pass in `self._total_threads` to here (fine, just slightly awkward).